src/layout/barGrid.ts

```diff
@@ -533,14 +538,18 @@ export function createProgressiveLayout(seriesType: string): StageHandler {
                                 const startCoord = cartesian.dataToPoint([startValue, baseValue]);
                                 baseCoord = startCoord[0];
                             }
-                            x = baseCoord;
+                            x = lastStackCoords[baseValue] || baseCoord;
```

Comment:
`lastStackCoords[baseValue]` could be 0 (e.g., `grid.left` is 0) so using `||` is not correct. Use `retrieve2` instead.